another sample for the problem in global variable definition

First thanks to you all. The following program `sample.go` triggers a panic(maybe related to #730 ):

```go
package main

import "strconv"

var optionsG map[string]string

var roundG int = 30

func strToInt(s string, defaultValue int) int {
	n, err := strconv.ParseInt(s, 10, 0)
	if err != nil {
		return defaultValue
	}

	return int(n)
}

func main() {

	optionsG := map[string]string{"round": "12", "b": "one"}

	roundG = strToInt(optionsG["round"], 50)

	println(roundG)

	println(optionsG)
}

```

Expected result:
```console
$ go run ./sample.go
12
0xc000039e38
```

Got:
```console
$ yaegi ./sample.go
sample.go:10:12: panic
sample.go:20:32: panic
reflect.Set: value of type int is not assignable to type map[string]string
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc000368600, 0x40)
        c:/go/src/runtime/debug/stack.go:24 +0xa4
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc00008fde0)
        D:/goprjs/src/github.com/containous/yaegi/interp/interp.go:333 +0xc3
panic(0xce3a40, 0xc00018ce90)
        c:/go/src/runtime/panic.go:969 +0x174
github.com/containous/yaegi/interp.runCfg.func1(0xc0003e40a0, 0xc0003f0100)
        D:/goprjs/src/github.com/containous/yaegi/interp/run.go:115 +0x213
panic(0xce3a40, 0xc00018ce90)
        c:/go/src/runtime/panic.go:969 +0x174
github.com/containous/yaegi/interp.runCfg.func1(0xc0003e4140, 0xc0003ee400)
        D:/goprjs/src/github.com/containous/yaegi/interp/run.go:115 +0x213
panic(0xce3a40, 0xc00018ce90)
        c:/go/src/runtime/panic.go:969 +0x174
reflect.Value.assignTo(0xce2f00, 0xc000402040, 0x182, 0xe1ae4b, 0xb, 0xd29d60, 0x0, 0xc000402050, 0x82, 0xc000402050)
        c:/go/src/reflect/value.go:2403 +0x42d
reflect.Value.Set(0xd29d60, 0xc000190288, 0x195, 0xce2f00, 0xc000402040, 0x182)
        c:/go/src/reflect/value.go:1532 +0xc4
github.com/containous/yaegi/interp._return.func1(0xc0003e4140, 0xc00018cce0)
        D:/goprjs/src/github.com/containous/yaegi/interp/run.go:1792 +0x88
github.com/containous/yaegi/interp.runCfg(0xc0003ee400, 0xc0003e4140)
        D:/goprjs/src/github.com/containous/yaegi/interp/run.go:121 +0x74
github.com/containous/yaegi/interp.call.func7(0xc0003e40a0, 0xc0003e1ab0)
        D:/goprjs/src/github.com/containous/yaegi/interp/run.go:991 +0xc49
github.com/containous/yaegi/interp.runCfg(0xc0003f0100, 0xc0003e40a0)
        D:/goprjs/src/github.com/containous/yaegi/interp/run.go:121 +0x74
github.com/containous/yaegi/interp.(*Interpreter).run(0xc0001b61a0, 0xc0003ef500, 0xc0003e4000)
        D:/goprjs/src/github.com/containous/yaegi/interp/run.go:99 +0x240
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0001b61a0, 0xc000364340, 0x19f, 0x0, 0x0, 0x0, 0x0, 0x0)
        D:/goprjs/src/github.com/containous/yaegi/interp/interp.go:409 +0x532
main.main()
        D:/goprjs/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:159 +0x664
```

--- a/_test/issue-735.go
+++ b/_test/issue-735.go
@@ -1,0 +1,29 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+)
+
+var optionsG map[string]string
+
+var roundG int = 30
+
+func strToInt(s string, defaultValue int) int {
+	n, err := strconv.ParseInt(s, 10, 0)
+	if err != nil {
+		return defaultValue
+	}
+	return int(n)
+}
+
+func main() {
+	optionsG := map[string]string{"round": "12", "b": "one"}
+	roundG = strToInt(optionsG["round"], 50)
+	fmt.Println(roundG)
+	fmt.Println(optionsG)
+}
+
+// Output:
+// 12
+// map[b:one round:12]

--- a/_test/time13.go
+++ b/_test/time13.go
@@ -1,0 +1,18 @@
+package main
+
+import (
+	"fmt"
+	"time"
+)
+
+var dummy = 1
+
+var t time.Time = time.Date(2007, time.November, 10, 23, 4, 5, 0, time.UTC)
+
+func main() {
+	t = time.Date(2009, time.November, 10, 23, 4, 5, 0, time.UTC)
+	fmt.Println(t.Clock())
+}
+
+// Output:
+// 23 4 5

--- a/interp/run.go
+++ b/interp/run.go
@@ -505,19 +505,21 @@ func deref(n *node) {
 	value := genValue(n.child[0])
 	tnext := getExec(n.tnext)
 	i := n.findex
+	l := n.level
 
 	if n.fnext != nil {
 		fnext := getExec(n.fnext)
 		n.exec = func(f *frame) bltn {
-			f.data[i] = value(f).Elem()
-			if f.data[i].Bool() {
+			r := value(f).Elem()
+			if r.Bool() {
+				getFrame(f, l).data[i] = r
 				return tnext
 			}
 			return fnext
 		}
 	} else {
 		n.exec = func(f *frame) bltn {
-			f.data[i] = value(f).Elem()
+			getFrame(f, l).data[i] = value(f).Elem()
 			return tnext
 		}
 	}
@@ -852,9 +854,10 @@ func call(n *node) {
 		}
 	default:
 		// Multiple return values frame index are indexed from the node frame index.
+		l := n.level
 		for i := range rtypes {
 			j := n.findex + i
-			rvalues[i] = func(f *frame) reflect.Value { return f.data[j] }
+			rvalues[i] = func(f *frame) reflect.Value { return getFrame(f, l).data[j] }
 		}
 	}
 
@@ -1006,6 +1009,21 @@ func pindex(i, variadic int) int {
 	return variadic
 }
 
+func getFrame(f *frame, l int) *frame {
+	switch l {
+	case 0:
+		return f
+	case 1:
+		return f.anc
+	case 2:
+		return f.anc.anc
+	}
+	for ; l > 0; l-- {
+		f = f.anc
+	}
+	return f
+}
+
 // Callbin calls a function from a bin import, accessible through reflect.
 func callBin(n *node) {
 	tnext := getExec(n.tnext)
@@ -1106,6 +1124,7 @@ func callBin(n *node) {
 	case fnext != nil:
 		// Handle branching according to boolean result.
 		index := n.findex
+		level := n.level
 		n.exec = func(f *frame) bltn {
 			in := make([]reflect.Value, l)
 			for i, v := range values {
@@ -1113,7 +1132,7 @@ func callBin(n *node) {
 			}
 			res := callFn(value(f), in)
 			b := res[0].Bool()
-			f.data[index].SetBool(b)
+			getFrame(f, level).data[index].SetBool(b)
 			if b {
 				return tnext
 			}
@@ -1169,7 +1188,7 @@ func callBin(n *node) {
 				}
 				out := callFn(value(f), in)
 				for i := 0; i < len(out); i++ {
-					f.data[n.findex+i].Set(out[i])
+					getFrame(f, n.level).data[n.findex+i].Set(out[i])
 				}
 				return tnext
 			}
@@ -1180,6 +1199,7 @@ func callBin(n *node) {
 func getIndexBinMethod(n *node) {
 	// dest := genValue(n)
 	i := n.findex
+	l := n.level
 	m := n.val.(int)
 	value := genValue(n.child[0])
 	next := getExec(n.tnext)
@@ -1187,20 +1207,21 @@ func getIndexBinMethod(n *node) {
 	n.exec = func(f *frame) bltn {
 		// Can not use .Set() because dest type contains the receiver and source not
 		// dest(f).Set(value(f).Method(m))
-		f.data[i] = value(f).Method(m)
+		getFrame(f, l).data[i] = value(f).Method(m)
 		return next
 	}
 }
 
 func getIndexBinPtrMethod(n *node) {
 	i := n.findex
+	l := n.level
 	m := n.val.(int)
 	value := genValue(n.child[0])
 	next := getExec(n.tnext)
 
 	n.exec = func(f *frame) bltn {
 		// Can not use .Set() because dest type contains the receiver and source not
-		f.data[i] = value(f).Addr().Method(m)
+		getFrame(f, l).data[i] = value(f).Addr().Method(m)
 		return next
 	}
 }
@@ -1210,21 +1231,23 @@ func getIndexArray(n *node) {
 	tnext := getExec(n.tnext)
 	value0 := genValueArray(n.child[0]) // array
 	i := n.findex
+	l := n.level
 
 	if n.child[1].rval.IsValid() { // constant array index
 		ai := int(vInt(n.child[1].rval))
 		if n.fnext != nil {
 			fnext := getExec(n.fnext)
 			n.exec = func(f *frame) bltn {
-				f.data[i] = value0(f).Index(ai)
-				if f.data[i].Bool() {
+				r := value0(f).Index(ai)
+				getFrame(f, l).data[i] = r
+				if r.Bool() {
 					return tnext
 				}
 				return fnext
 			}
 		} else {
 			n.exec = func(f *frame) bltn {
-				f.data[i] = value0(f).Index(ai)
+				getFrame(f, l).data[i] = value0(f).Index(ai)
 				return tnext
 			}
 		}
@@ -1235,8 +1258,9 @@ func getIndexArray(n *node) {
 			fnext := getExec(n.fnext)
 			n.exec = func(f *frame) bltn {
 				_, vi := value1(f)
-				f.data[i] = value0(f).Index(int(vi))
-				if f.data[i].Bool() {
+				r := value0(f).Index(int(vi))
+				getFrame(f, l).data[i] = r
+				if r.Bool() {
 					return tnext
 				}
 				return fnext
@@ -1244,7 +1268,7 @@ func getIndexArray(n *node) {
 		} else {
 			n.exec = func(f *frame) bltn {
 				_, vi := value1(f)
-				f.data[i] = value0(f).Index(int(vi))
+				getFrame(f, l).data[i] = value0(f).Index(int(vi))
 				return tnext
 			}
 		}
@@ -1447,6 +1471,7 @@ func getFunc(n *node) {
 
 func getMethod(n *node) {
 	i := n.findex
+	l := n.level
 	next := getExec(n.tnext)
 
 	n.exec = func(f *frame) bltn {
@@ -1455,7 +1480,7 @@ func getMethod(n *node) {
 		nod.val = &nod
 		nod.recv = n.recv
 		nod.frame = fr
-		f.data[i] = reflect.ValueOf(&nod)
+		getFrame(f, l).data[i] = reflect.ValueOf(&nod)
 		return next
 	}
 }
@@ -1465,6 +1490,7 @@ func getMethodByName(n *node) {
 	value0 := genValue(n.child[0])
 	name := n.child[1].ident
 	i := n.findex
+	l := n.level
 
 	n.exec = func(f *frame) bltn {
 		val := value0(f).Interface().(valueInterface)
@@ -1474,7 +1500,7 @@ func getMethodByName(n *node) {
 		nod.val = &nod
 		nod.recv = &receiver{nil, val.value, li}
 		nod.frame = fr
-		f.data[i] = reflect.ValueOf(&nod)
+		getFrame(f, l).data[i] = reflect.ValueOf(&nod)
 		return next
 	}
 }
@@ -1484,6 +1510,7 @@ func getIndexSeq(n *node) {
 	index := n.val.([]int)
 	tnext := getExec(n.tnext)
 	i := n.findex
+	l := n.level
 
 	// Note:
 	// Here we have to store the result using
@@ -1498,15 +1525,16 @@ func getIndexSeq(n *node) {
 	if n.fnext != nil {
 		fnext := getExec(n.fnext)
 		n.exec = func(f *frame) bltn {
-			f.data[i] = value(f).FieldByIndex(index)
-			if f.data[i].Bool() {
+			r := value(f).FieldByIndex(index)
+			getFrame(f, l).data[i] = r
+			if r.Bool() {
 				return tnext
 			}
 			return fnext
 		}
 	} else {
 		n.exec = func(f *frame) bltn {
-			f.data[i] = value(f).FieldByIndex(index)
+			getFrame(f, l).data[i] = value(f).FieldByIndex(index)
 			return tnext
 		}
 	}
@@ -1523,19 +1551,21 @@ func getPtrIndexSeq(n *node) {
 		value = genValue(n.child[0])
 	}
 	i := n.findex
+	l := n.level
 
 	if n.fnext != nil {
 		fnext := getExec(n.fnext)
 		n.exec = func(f *frame) bltn {
-			f.data[i] = value(f).Elem().FieldByIndex(index)
-			if f.data[i].Bool() {
+			r := value(f).Elem().FieldByIndex(index)
+			getFrame(f, l).data[i] = r
+			if r.Bool() {
 				return tnext
 			}
 			return fnext
 		}
 	} else {
 		n.exec = func(f *frame) bltn {
-			f.data[i] = value(f).Elem().FieldByIndex(index)
+			getFrame(f, l).data[i] = value(f).Elem().FieldByIndex(index)
 			return tnext
 		}
 	}
@@ -1545,22 +1575,25 @@ func getIndexSeqField(n *node) {
 	value := genValue(n.child[0])
 	index := n.val.([]int)
 	i := n.findex
+	l := n.level
 	tnext := getExec(n.tnext)
 
 	if n.fnext != nil {
 		fnext := getExec(n.fnext)
 		if n.child[0].typ.TypeOf().Kind() == reflect.Ptr {
 			n.exec = func(f *frame) bltn {
-				f.data[i] = value(f).Elem().FieldByIndex(index)
-				if f.data[i].Bool() {
+				r := value(f).Elem().FieldByIndex(index)
+				getFrame(f, l).data[i] = r
+				if r.Bool() {
 					return tnext
 				}
 				return fnext
 			}
 		} else {
 			n.exec = func(f *frame) bltn {
-				f.data[i] = value(f).FieldByIndex(index)
-				if f.data[i].Bool() {
+				r := value(f).FieldByIndex(index)
+				getFrame(f, l).data[i] = r
+				if r.Bool() {
 					return tnext
 				}
 				return fnext
@@ -1569,12 +1602,12 @@ func getIndexSeqField(n *node) {
 	} else {
 		if n.child[0].typ.TypeOf().Kind() == reflect.Ptr {
 			n.exec = func(f *frame) bltn {
-				f.data[i] = value(f).Elem().FieldByIndex(index)
+				getFrame(f, l).data[i] = value(f).Elem().FieldByIndex(index)
 				return tnext
 			}
 		} else {
 			n.exec = func(f *frame) bltn {
-				f.data[i] = value(f).FieldByIndex(index)
+				getFrame(f, l).data[i] = value(f).FieldByIndex(index)
 				return tnext
 			}
 		}
@@ -1587,16 +1620,17 @@ func getIndexSeqPtrMethod(n *node) {
 	fi := index[1:]
 	mi := index[0]
 	i := n.findex
+	l := n.level
 	next := getExec(n.tnext)
 
 	if n.child[0].typ.TypeOf().Kind() == reflect.Ptr {
 		n.exec = func(f *frame) bltn {
-			f.data[i] = value(f).Elem().FieldByIndex(fi).Addr().Method(mi)
+			getFrame(f, l).data[i] = value(f).Elem().FieldByIndex(fi).Addr().Method(mi)
 			return next
 		}
 	} else {
 		n.exec = func(f *frame) bltn {
-			f.data[i] = value(f).FieldByIndex(fi).Addr().Method(mi)
+			getFrame(f, l).data[i] = value(f).FieldByIndex(fi).Addr().Method(mi)
 			return next
 		}
 	}
@@ -1608,16 +1642,17 @@ func getIndexSeqMethod(n *node) {
 	fi := index[1:]
 	mi := index[0]
 	i := n.findex
+	l := n.level
 	next := getExec(n.tnext)
 
 	if n.child[0].typ.TypeOf().Kind() == reflect.Ptr {
 		n.exec = func(f *frame) bltn {
-			f.data[i] = value(f).Elem().FieldByIndex(fi).Method(mi)
+			getFrame(f, l).data[i] = value(f).Elem().FieldByIndex(fi).Method(mi)
 			return next
 		}
 	} else {
 		n.exec = func(f *frame) bltn {
-			f.data[i] = value(f).FieldByIndex(fi).Method(mi)
+			getFrame(f, l).data[i] = value(f).FieldByIndex(fi).Method(mi)
 			return next
 		}
 	}
@@ -1995,6 +2030,7 @@ func doCompositeLit(n *node, hasType bool) {
 	}
 
 	i := n.findex
+	l := n.level
 	n.exec = func(f *frame) bltn {
 		a := reflect.New(n.typ.TypeOf()).Elem()
 		for i, v := range values {
@@ -2006,7 +2042,7 @@ func doCompositeLit(n *node, hasType bool) {
 		case destInterface:
 			d.Set(reflect.ValueOf(valueInterface{n, a}))
 		default:
-			f.data[i] = a
+			getFrame(f, l).data[i] = a
 		}
 		return next
 	}
@@ -2582,6 +2618,7 @@ func recv(n *node) {
 	value := genValue(n.child[0])
 	tnext := getExec(n.tnext)
 	i := n.findex
+	l := n.level
 
 	if n.interp.cancelChan {
 		// Cancellable channel read
@@ -2589,10 +2626,10 @@ func recv(n *node) {
 			fnext := getExec(n.fnext)
 			n.exec = func(f *frame) bltn {
 				// Fast: channel read doesn't block
-				var ok bool
 				ch := value(f)
-				if f.data[i], ok = ch.TryRecv(); ok {
-					if f.data[i].Bool() {
+				if r, ok := ch.TryRecv(); ok {
+					getFrame(f, l).data[i] = r
+					if r.Bool() {
 						return tnext
 					}
 					return fnext
@@ -2610,14 +2647,14 @@ func recv(n *node) {
 		} else {
 			n.exec = func(f *frame) bltn {
 				// Fast: channel read doesn't block
-				var ok bool
 				ch := value(f)
-				if f.data[i], ok = ch.TryRecv(); ok {
+				if r, ok := ch.TryRecv(); ok {
+					getFrame(f, l).data[i] = r
 					return tnext
 				}
 				// Slow: channel is blocked, allow cancel
 				var chosen int
-				chosen, f.data[i], _ = reflect.Select([]reflect.SelectCase{f.done, {Dir: reflect.SelectRecv, Chan: ch}})
+				chosen, getFrame(f, l).data[i], _ = reflect.Select([]reflect.SelectCase{f.done, {Dir: reflect.SelectRecv, Chan: ch}})
 				if chosen == 0 {
 					return nil
 				}
@@ -2629,7 +2666,8 @@ func recv(n *node) {
 		if n.fnext != nil {
 			fnext := getExec(n.fnext)
 			n.exec = func(f *frame) bltn {
-				if f.data[i], _ = value(f).Recv(); f.data[i].Bool() {
+				if r, _ := value(f).Recv(); r.Bool() {
+					getFrame(f, l).data[i] = r
 					return tnext
 				}
 				return fnext
@@ -2637,7 +2675,7 @@ func recv(n *node) {
 		} else {
 			i := n.findex
 			n.exec = func(f *frame) bltn {
-				f.data[i], _ = value(f).Recv()
+				getFrame(f, l).data[i], _ = value(f).Recv()
 				return tnext
 			}
 		}
@@ -2918,6 +2956,7 @@ func _select(n *node) {
 // slice expression: array[low:high:max].
 func slice(n *node) {
 	i := n.findex
+	l := n.level
 	next := getExec(n.tnext)
 	value0 := genValueArray(n.child[0]) // array
 	value1 := genValue(n.child[1])      // low (if 2 or 3 args) or high (if 1 arg)
@@ -2926,7 +2965,7 @@ func slice(n *node) {
 	case 2:
 		n.exec = func(f *frame) bltn {
 			a := value0(f)
-			f.data[i] = a.Slice(int(value1(f).Int()), a.Len())
+			getFrame(f, l).data[i] = a.Slice(int(value1(f).Int()), a.Len())
 			return next
 		}
 	case 3:
@@ -2934,7 +2973,7 @@ func slice(n *node) {
 
 		n.exec = func(f *frame) bltn {
 			a := value0(f)
-			f.data[i] = a.Slice(int(value1(f).Int()), int(value2(f).Int()))
+			getFrame(f, l).data[i] = a.Slice(int(value1(f).Int()), int(value2(f).Int()))
 			return next
 		}
 	case 4:
@@ -2943,7 +2982,7 @@ func slice(n *node) {
 
 		n.exec = func(f *frame) bltn {
 			a := value0(f)
-			f.data[i] = a.Slice3(int(value1(f).Int()), int(value2(f).Int()), int(value3(f).Int()))
+			getFrame(f, l).data[i] = a.Slice3(int(value1(f).Int()), int(value2(f).Int()), int(value3(f).Int()))
 			return next
 		}
 	}
@@ -2952,6 +2991,7 @@ func slice(n *node) {
 // slice expression, no low value: array[:high:max].
 func slice0(n *node) {
 	i := n.findex
+	l := n.level
 	next := getExec(n.tnext)
 	value0 := genValueArray(n.child[0])
 
@@ -2959,14 +2999,14 @@ func slice0(n *node) {
 	case 1:
 		n.exec = func(f *frame) bltn {
 			a := value0(f)
-			f.data[i] = a.Slice(0, a.Len())
+			getFrame(f, l).data[i] = a.Slice(0, a.Len())
 			return next
 		}
 	case 2:
 		value1 := genValue(n.child[1])
 		n.exec = func(f *frame) bltn {
 			a := value0(f)
-			f.data[i] = a.Slice(0, int(value1(f).Int()))
+			getFrame(f, l).data[i] = a.Slice(0, int(value1(f).Int()))
 			return next
 		}
 	case 3:
@@ -2974,7 +3014,7 @@ func slice0(n *node) {
 		value2 := genValue(n.child[2])
 		n.exec = func(f *frame) bltn {
 			a := value0(f)
-			f.data[i] = a.Slice3(0, int(value1(f).Int()), int(value2(f).Int()))
+			getFrame(f, l).data[i] = a.Slice3(0, int(value1(f).Int()), int(value2(f).Int()))
 			return next
 		}
 	}

